### PR TITLE
added logging boolean for mig healthcheck

### PR DESCRIPTION
**Pull request by hjoh95**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_generate_modules
 The modules need to be regenerated. Please run `make_build`. 
 ```bash 
Checking submodule's files generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/mig/README.md /tmp/tmp.dDesyM8wgg/workspace/modules/mig/README.md
30c30
< | health\_check | Health check to determine whether instances are responsive and able to do work | <pre>object({<br>    type                = string<br>    initial_delay_sec   = number<br>    check_interval_sec  = number<br>    healthy_threshold   = number<br>    timeout_sec         = number<br>    unhealthy_threshold = number<br>    response            = string<br>    proxy_header        = string<br>    port                = number<br>    request             = string<br>    request_path        = string<br>    host                = string<br>    enable_logging      = bool<br>  })</pre> | <pre>{<br>  "check_interval_sec": 30,<br>  "healthy_threshold": 1,<br>  "host": "",<br>  "initial_delay_sec": 30,<br>  "port": 80,<br>  "proxy_header": "NONE",<br>  "request": "",<br>  "request_path": "/",<br>  "response": "",<br>  "timeout_sec": 10,<br>  "type": "",<br>  "unhealthy_threshold": 5,<br>  "enable_logging": false}</pre> | no |
---
> | health\_check | Health check to determine whether instances are responsive and able to do work | <pre>object({<br>    type                = string<br>    initial_delay_sec   = number<br>    check_interval_sec  = number<br>    healthy_threshold   = number<br>    timeout_sec         = number<br>    unhealthy_threshold = number<br>    response            = string<br>    proxy_header        = string<br>    port                = number<br>    request             = string<br>    request_path        = string<br>    host                = string<br>  })</pre> | <pre>{<br>  "check_interval_sec": 30,<br>  "healthy_threshold": 1,<br>  "host": "",<br>  "initial_delay_sec": 30,<br>  "port": 80,<br>  "proxy_header": "NONE",<br>  "request": "",<br>  "request_path": "/",<br>  "response": "",<br>  "timeout_sec": 10,<br>  "type": "",<br>  "unhealthy_threshold": 5<br>}</pre> | no |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/mig/main.tf /tmp/tmp.dDesyM8wgg/workspace/modules/mig/main.tf
181,183d180
<   log_config {
<     enable = var.health_check["enable_logging"]
<   }
203,205d199
<   log_config {
<     enable = var.health_check["enable_logging"]
<   }
224,227c218
<   log_config {
<     enable = var.health_check["enable_logging"]
<   }
< }
\ No newline at end of file
---
> }
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/mig/variables.tf /tmp/tmp.dDesyM8wgg/workspace/modules/mig/variables.tf
118d117
<     enable_logging      = bool
133d131
<     enable_logging      = false
Error: submodule's files generation has not been run, please run the
'make build' command and commit changes
 ``` 
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/mig/README.md /tmp/tmp.oHcINwCGFJ/generate_docs/workspace/modules/mig/README.md
30c30
< | health\_check | Health check to determine whether instances are responsive and able to do work | <pre>object({<br>    type                = string<br>    initial_delay_sec   = number<br>    check_interval_sec  = number<br>    healthy_threshold   = number<br>    timeout_sec         = number<br>    unhealthy_threshold = number<br>    response            = string<br>    proxy_header        = string<br>    port                = number<br>    request             = string<br>    request_path        = string<br>    host                = string<br>    enable_logging      = bool<br>  })</pre> | <pre>{<br>  "check_interval_sec": 30,<br>  "healthy_threshold": 1,<br>  "host": "",<br>  "initial_delay_sec": 30,<br>  "port": 80,<br>  "proxy_header": "NONE",<br>  "request": "",<br>  "request_path": "/",<br>  "response": "",<br>  "timeout_sec": 10,<br>  "type": "",<br>  "unhealthy_threshold": 5,<br>  "enable_logging": false}</pre> | no |
---
> | health\_check | Health check to determine whether instances are responsive and able to do work | <pre>object({<br>    type                = string<br>    initial_delay_sec   = number<br>    check_interval_sec  = number<br>    healthy_threshold   = number<br>    timeout_sec         = number<br>    unhealthy_threshold = number<br>    response            = string<br>    proxy_header        = string<br>    port                = number<br>    request             = string<br>    request_path        = string<br>    host                = string<br>    enable_logging      = bool<br>  })</pre> | <pre>{<br>  "check_interval_sec": 30,<br>  "enable_logging": false,<br>  "healthy_threshold": 1,<br>  "host": "",<br>  "initial_delay_sec": 30,<br>  "port": 80,<br>  "proxy_header": "NONE",<br>  "request": "",<br>  "request_path": "/",<br>  "response": "",<br>  "timeout_sec": 10,<br>  "type": "",<br>  "unhealthy_threshold": 5<br>}</pre> | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
Checking for missing newline at end of file
Error: No newline at end of file ./modules/mig/main.tf
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate ./examples/compute_instance/disk_snapshot 
Success!
 The configuration is valid.
terraform_validate ./examples/compute_instance/multiple_interfaces 
Success!
 The configuration is valid.
terraform_validate ./examples/compute_instance/next_hop 
Success!
 The configuration is valid.
terraform_validate ./examples/compute_instance/simple 
Success!
 The configuration is valid.
terraform_validate ./examples/compute_instance/tags 
Success!
 The configuration is valid.
terraform_validate ./examples/instance_template/additional_disks 
Success!
 The configuration is valid.
terraform_validate ./examples/instance_template/alias_ip_range 
Success!
 The configuration is valid.
terraform_validate ./examples/instance_template/encrypted_disks 
Success!
 The configuration is valid.
terraform_validate ./examples/instance_template/simple 
Success!
 The configuration is valid.
terraform_validate ./examples/mig/autoscaler 
Success!
 The configuration is valid.
terraform_validate ./examples/mig/full 
╷
│
 
Error: 
Invalid value for module argument
│
 
│
 
  on main.tf line 72, in module "mig":
│
 
  72:   health_check = 
var.health_check
│
 
│
 
The given value is not suitable for child module variable "health_check"
│
 
defined at ../../../modules/mig/variables.tf:103,1-24: attribute
│
 
"enable_logging" is required.
╵
terraform_validate ./examples/mig/healthcheck 
╷
│
 
Error: 
Invalid value for module argument
│
 
│
 
  on main.tf line 79, in module "mig":
│
 
  79:   health_check = 
{
│
 
  80:     type                = "https"
│
 
  81:     initial_delay_sec   = 120
│
 
  82:     check_interval_sec  = 5
│
 
  83:     healthy_threshold   = 2
│
 
  84:     timeout_sec         = 5
│
 
  85:     unhealthy_threshold = 2
│
 
  86:     response            = ""
│
 
  87:     proxy_header        = "NONE"
│
 
  88:     port                = 443
│
 
  89:     request             = ""
│
 
  90:     request_path        = "/"
│
 
  91:     host                = "localhost"
│
 
  92:   }
│
 
│
 
The given value is not suitable for child module variable "health_check"
│
 
defined at ../../../modules/mig/variables.tf:103,1-24: attribute
│
 
"enable_logging" is required.
╵
terraform_validate ./examples/mig/simple 
Success!
 The configuration is valid.
terraform_validate ./examples/mig_with_percent/simple 
Success!
 The configuration is valid.
terraform_validate ./examples/preemptible_and_regular_instance_templates/simple 
Success!
 The configuration is valid.
terraform_validate ./examples/umig/full 
Success!
 The configuration is valid.
terraform_validate ./examples/umig/named_ports 
Success!
 The configuration is valid.
terraform_validate ./examples/umig/simple 
Success!
 The configuration is valid.
terraform_validate ./examples/umig/static_ips 
Success!
 The configuration is valid.
terraform_validate ./modules/compute_disk_snapshot 
Success!
 The configuration is valid.
terraform_validate ./modules/compute_instance 
Success!
 The configuration is valid.
terraform_validate ./modules/instance_template 
Success!
 The configuration is valid.
terraform_validate ./modules/mig 
Success!
 The configuration is valid.
terraform_validate ./modules/mig_with_percent 
Success!
 The configuration is valid.
terraform_validate ./modules/preemptible_and_regular_instance_templates 
Success!
 The configuration is valid.
terraform_validate ./modules/umig 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/compute_instance/disk_snapshot 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/compute_instance/simple_zone 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/instance_simple 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/instance_template/additional_disks 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/instance_template/alias_ip_range 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/instance_template/simple 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/mig/autoscaler 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/mig/healthcheck 
╷
│
 
Error: 
Invalid value for module argument
│
 
│
 
  on ../../../../examples/mig/healthcheck/main.tf line 79, in module "mig":
│
 
  79:   health_check = 
{
│
 
  80:     type                = "https"
│
 
  81:     initial_delay_sec   = 120
│
 
  82:     check_interval_sec  = 5
│
 
  83:     healthy_threshold   = 2
│
 
  84:     timeout_sec         = 5
│
 
  85:     unhealthy_threshold = 2
│
 
  86:     response            = ""
│
 
  87:     proxy_header        = "NONE"
│
 
  88:     port                = 443
│
 
  89:     request             = ""
│
 
  90:     request_path        = "/"
│
 
  91:     host                = "localhost"
│
 
  92:   }
│
 
│
 
The given value is not suitable for child module variable "health_check"
│
 
defined at ../../../../modules/mig/variables.tf:103,1-24: attribute
│
 
"enable_logging" is required.
╵
terraform_validate ./test/fixtures/mig/simple 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/mig_with_percent/simple 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/preemptible_and_regular_instance_templates/simple 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/umig/named_ports 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/umig/simple 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/fixtures/umig/static_ips 
╷
│
 
Warning: 
Version constraints inside provider configuration blocks are deprecated
│
 
│
 
  on network.tf line 18, in provider "random":
│
 
  18:   version = 
"~> 2.2"
│
 
│
 
Terraform 0.13 and earlier allowed provider version constraints inside the
│
 
provider configuration block, but that is now deprecated and will be
│
 
removed in a future version of Terraform. To silence this warning, move the
│
 
provider version constraint into the required_providers block.
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/setup 
Success!
 The configuration is valid.
 ```

**Comment by hjoh95** [comment]
Hoping to get some help here as I'm unfamiliar with using `makefile`. I've run `make build` and `make generate_docs` from the root `/terraform-google-vm` directory, which seem to complete, but no files are changed. My assumption is that the issue is because my terraform workspace (and git, make, etc)  is a docker container itself, so `make build` is attempting to create a nested container.

I am also confused by `check_terraform` because I don't know where to edit the inputs to include `enable_logging`

File: modules/mig/main.tf [hunk 1]
```diff
@@ -197,6 +200,9 @@ resource "google_compute_health_check" "http" {
     response     = var.health_check["response"]
     proxy_header = var.health_check["proxy_header"]
   }
+  log_config {
```
Comment (bharathkkb) [inline modules/mig/main.tf:203]: Looks like this was added in 3.71 https://github.com/hashicorp/terraform-provider-google/blob/v3.90.1/CHANGELOG.md#3710-june-07-2021
Can we bump the provider version here https://github.com/terraform-google-modules/terraform-google-vm/blob/28c77b553d1f4462ef21fe55f4e4bfb26eb6d892/autogen/versions.tf.tmpl#L20-L22

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR. The modules/mig is autogenerated so the change should be made in https://github.com/terraform-google-modules/terraform-google-vm/tree/master/autogen

**Pull request closed without merging**